File: C_Sharp_code/Example/WindowsFormsApp/VB_demo10/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim r As Integer = New Random().Next(1, 101)
        Dim guess As Integer
        Dim flag As Boolean
        For i As Integer = 1 To 3
            guess = InputBox("請輸入要猜的數字1~100")
            If (guess > r) Then
                MessageBox.Show("Lower")
            ElseIf (guess < r) Then
                MessageBox.Show("Higher")
            Else
                flag = True
                Exit For
            End If
        Next
        If (flag) Then
            MessageBox.Show("猜中了!" & guess)
        Else
            MessageBox.Show("沒猜中!" & r)

        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Dim r As Integer = New Random().Next(1, 101)
        Dim guess As Integer
        Dim i As Integer = 1
        Dim flag As Boolean
        Do While (i <= 3)
            guess = InputBox("請輸入要猜的數字1~100")
            If (guess > r) Then
                MessageBox.Show("Lower")
            ElseIf (guess < r) Then
                MessageBox.Show("Higher")
            Else
                flag = True
                Exit Do
            End If
            i += 1
        Loop
        If (flag) Then
            MessageBox.Show("猜中了!" & guess)
        Else
            MessageBox.Show("沒猜中!" & r)
        End If
    End Sub
End Class

File: C_Sharp_code/Example/WindowsFormsApp/CS_demo10/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace CS_demo10
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            int r = new Random().Next(1, 101);
            int guess = 0;
            bool flag = false;
            for (int i = 1; i <= 3; i++)
            {
                guess = Convert.ToInt32(Interaction.InputBox("請輸入數字1~100", "視窗", "50"));
                if (guess > r)
                {
                    MessageBox.Show("Lower");
                }
                else if (guess < r)
                {
                    MessageBox.Show("Higher");
                }
                else
                {
                    flag = true;
                    break;
                }
            }
            if (flag)
                MessageBox.Show("猜中了!" + guess);
            else
                MessageBox.Show("沒猜中!" + r);
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            int r = new Random().Next(1, 101);
            int guess = 0;
            //bool flag = false;
            //int i = 1;
            do
            {
                guess = Convert.ToInt32(Interaction.InputBox("請輸入數字1~100"));
                if (guess > r)
                {
                    MessageBox.Show("Lower");
                }
                else if (guess < r)
                {
                    MessageBox.Show("Higher");
                }
                else
                {
                    //flag = true;
                    break;
                }
                //i ++;
            } while (true);
            //if (flag)
            //    MessageBox.Show("猜中了!" + guess);
            // else
            //    MessageBox.Show("沒猜中!" + r);
            MessageBox.Show("猜中了!" + r);
        }
    }

}
